### PR TITLE
Add duplicateLetter exercise + implementation

### PR DESCRIPTION
**Pull request by vaibhavarora14**
**Program Strategy -**
To find out most duplicate letter `count` from a `string`, program does following things:
- With `Set`, unique letters copy is pulled out from `string`.
- For every `unique letter`, a filter runs on 'string' to find out that `letter` length.
    - If letter length is greater than, we found previously, it's `stored`.
    - With this, `stored` count in end is count of maximum duplicate `letter count` from `string`.

### CODE REVIEW

**Comment by varora1406** [comment]
Performance of program is not good according to me.
Working on it's performance.

**Comment by varora1406** [comment]
Program is ready for review

File: js-exercises/duplicate-letters/duplicateLetters.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      throw Error(`Expects english characters only, and input has ${letter}`);
+    }
+
+    const letterCount = letters.filter(
+      (filteredLetter) => filteredLetter === letter
+    ).length;
+
+    if (letterCount > mostDuplicateLetter.count) {
+      mostDuplicateLetter.count = letterCount;
+      mostDuplicateLetter.letter = letter;
```
Comment (ahhda) [inline js-exercises/duplicate-letters/duplicateLetters.js:38]: Do we really need to store the letter as well? Just trying to understand your thought process. Were you trying to implement thinking that in the future you might need it?

Comment (varora1406) [reply js-exercises/duplicate-letters/duplicateLetters.js:38]: This program just needed count, but I went with real time expectation to keep letter as well in program.
Will remove from here for the time, and keep it fully according to specs

Comment (varora1406) [reply js-exercises/duplicate-letters/duplicateLetters.js]: it's removed with https://github.com/pesto-students/deep-dive-varora1406/pull/1/commits/f47a8012116b60b63ef809392c344e14ec1e118f

File: js-exercises/duplicate-letters/duplicateLetters.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const isEmptyString = (argument) => {
+  if (typeof argument !== 'string') {
+    throw TypeError(`expects string input, and input has ${typeof argument}`);
+  }
+
+  if (argument.trim().length < 1) {
+    return true;
+  }
+
+  return false;
```
[the remark this answers is not in the document]
Comment (varora1406) [reply js-exercises/duplicate-letters/duplicateLetters.js]: improved with https://github.com/pesto-students/deep-dive-varora1406/pull/1/commits/47178678036add39e53c5f851b743f4bd3f29b3d

File: js-exercises/duplicate-letters/duplicateLetters.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+const letterRegex = /^[A-Za-z]+$/;
+
+const isLetter = (letter) => letterRegex.test(letter);
```
[the remark this answers is not in the document]
Comment (varora1406) [reply js-exercises/duplicate-letters/duplicateLetters.js:11]: Actually by `letter`, I meant an `alphabet`.
So letterRegex should neither be true for `abc def` nor even it should be true for `abc` OR `def`.

With this, I realize isLetter currently not handling length of letters. will update this in solution soon

File: js-exercises/duplicate-letters/duplicateLetters.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      mostDuplicateLetter.count = letterCount;
+      mostDuplicateLetter.letter = letter;
+    }
+  }
+
+  if (mostDuplicateLetter.count < 2) {
+    return false;
+  }
+
+  return mostDuplicateLetter.count;
```
[the remark this answers is not in the document]
Comment (varora1406) [reply js-exercises/duplicate-letters/duplicateLetters.js]: just wanted to make code easily readable.
If we can ignore LOC, and you can share the way you would like to see the code for your eyes.

 I will be happy to receive that feedback. 🙂

File: js-exercises/duplicate-letters/duplicateLetters.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (varora1406) [reply js-exercises/duplicate-letters/duplicateLetters.js:11]: it's available with 5b203a656feb2bfee9b9d66bfe59c9d8fdcd9937

File: js-exercises/duplicate-letters/duplicateLetters.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  const letters = [...argument];
+  const uniqueLetters = [...new Set(letters)];
+
+  const mostDuplicateLetter = { count: 0, letter: null };
+
+  for (const letter of uniqueLetters) {
+    if (!isLetter(letter)) {
+      mostDuplicateLetter.count = 0;
+      mostDuplicateLetter.letter = null;
+      break;
```
[the remark this answers is not in the document]
Comment (varora1406) [reply js-exercises/duplicate-letters/duplicateLetters.js]: Yes, with latest code.
The `if` block now throws error actually. so both of our code block won't fit there now 😃.
Have a look at it 😉

File: js-exercises/duplicate-letters/duplicateLetters.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (1akshat) [reply js-exercises/duplicate-letters/duplicateLetters.js]: yes definitely understand that.

File: js-exercises/duplicate-letters/duplicateLetters.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (1akshat) [reply js-exercises/duplicate-letters/duplicateLetters.js]: sure.

**Comment by varora1406** [comment]
Guys, I have updated subject with code strategy used in this program

**Pull request closed without merging**